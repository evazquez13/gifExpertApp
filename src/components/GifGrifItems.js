import React from 'react';
import PropTypes from 'prop-types';

export const GifGrifItems = ({title,url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGrifItems.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired    
} 