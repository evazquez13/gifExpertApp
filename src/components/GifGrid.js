import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGrifItems } from './GifGrifItems';

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDownBig">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading</p>}
            
            <div className="card-grid animate__animated animate__flipInY">
                {
                    data.map( img => (<GifGrifItems
                                            key={img.id}
                                            {...img}/>))
                }
                
            </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired
}
