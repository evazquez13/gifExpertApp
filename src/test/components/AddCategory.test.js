import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas al componente <AddCategory/>', () =>{

    const setCategory = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategory}/>);

    beforeEach( () =>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategory}/>);
    });

    test('Validar que el componente se muestre correctamente ', () =>{
        expect(wrapper).toMatchSnapshot();
    });

    test('Evaluar cambios de texto en el input', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {target:{value} });
    });

    test('No debe postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        //Valida que la funcion no se ejecute
        expect(setCategory).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () =>{
        const value = 'Hola mundo';
        wrapper.find('input').simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        
        //Valida que la funcion se ejecute
        expect(setCategory).toHaveBeenCalled();
        //Valida que la funcion se ejecute n cantidad de veces como se ponga dentro del parentesis
        expect(setCategory).toHaveBeenCalledTimes(1);
        //Valida que se llame una funcion
        expect(setCategory).toBeCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');
    });

});