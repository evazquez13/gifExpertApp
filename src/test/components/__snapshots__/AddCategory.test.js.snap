// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <AddCategory/> Validar que el componente se muestre correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
